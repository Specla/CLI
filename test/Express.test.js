/* eslint-env jest */
import Express from '../src/Express'
import setup from './utils/setup'
import cleanup from './utils/cleanup'

beforeEach(setup)
afterEach(cleanup)

test('Should export the enhanced Router class', () => {
  expect(!Express.Router).toBe(true)
})

test('Should export properties', () => {
  const app = new Express()
  expect(app.locals).toMatchSnapshot()
  expect(app.mountpath).toMatchSnapshot()
})

test('Should proxy .use to the express instance', () => {
  const app = new Express()
  app.use(function testMiddleware (req, res, next) {})
  expect(app._express._router.stack[2].name).toBe('testMiddleware')
})

test('Should proxy .all to the express instance', () => {
  const app = new Express()
  app.all('all', function allCallback (req, res) {})
  expect(app._express._router.stack[2]).toMatchSnapshot()
})

test('Should proxy .get (http) to the express instance', () => {
  const app = new Express()
  app.get('get', function getCallback (req, res) {})
  expect(app._express._router.stack[2]).toMatchSnapshot()
})

test('Should proxy .get (retreive value) to the express instance', () => {
  const app = new Express()
  app.enable('trust proxy')
  expect(app.get('trust proxy')).toBe(true)
})

test('Should proxy .post to the express instance', () => {
  const app = new Express()
  app.post('post', function deleteCallback (req, res) {})
  expect(app._express._router.stack[2]).toMatchSnapshot()
})

test('Should proxy .put to the express instance', () => {
  const app = new Express()
  app.put('put', function deleteCallback (req, res) {})
  expect(app._express._router.stack[2]).toMatchSnapshot()
})

test('Should proxy .delete to the express instance', () => {
  const app = new Express()
  app.delete('delete', function deleteCallback (req, res) {})
  expect(app._express._router.stack[2]).toMatchSnapshot()
})

test('Should proxy .disable to the express instance', () => {
  const app = new Express()
  app.disable('trust proxy')
  expect(app._express.get('trust proxy')).toBe(false)
})

test('Should proxy .disabled to the express instance', () => {
  const app = new Express()
  app.disable('trust proxy')
  expect(app.disabled('trust proxy')).toBe(true)
})

test('Should proxy .enable to the express instance', () => {
  const app = new Express()
  app.enable('trust proxy')
  expect(app._express.enabled('trust proxy')).toBe(true)
})

test('Should proxy .enabled to the express instance', () => {
  const app = new Express()
  app.enable('trust proxy')
  expect(app.enabled('trust proxy')).toBe(true)
})

test('Should proxy .engine to the express instance', () => {
  const app = new Express()
  app.engine('pug', () => {})
  expect(app._express.engines).toMatchSnapshot()
})

test('Should proxy .listen to the express instance', done => {
  const app = new Express()
  const server = app.listen(3000, () => {
    server.close()
    done()
  })
})

test('Should proxy .param to the express instance', () => {
  const app = new Express()
  app.param('id', () => {})
  expect(app._express._router.params).toMatchSnapshot()
})

test('Should proxy .path to the express instance', () => {
  const app = new Express()
  const blog = new Express()
  app.use('/test', blog)
  expect(blog.path()).toBe('/test')
})

test('Should proxy .render method to express instance', () => {
  const app = new Express()
  expect(app.render instanceof Function).toBe(true)
})

test('Should proxy .route method to express instance', () => {
  const app = new Express()
  app.route('/test')
    .all(function all () {})
    .get(function get () {})

  expect(app._express._router.stack[2]).toMatchSnapshot()
})

test('Should proxy .set method to express instance', () => {
  const app = new Express()
  app.set('test', true)
  expect(app.get('test')).toBe(true)
})

test('Should proxy .on method to the express instance', () => {
  const app = new Express()
  app.on('mount', () => {})
  expect(app._express._events).toMatchSnapshot()
})
