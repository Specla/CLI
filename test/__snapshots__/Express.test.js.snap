// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should export properties 1`] = `
Object {
  "settings": Object {
    "env": "test",
    "etag": "weak",
    "etag fn": [Function],
    "jsonp callback name": "callback",
    "query parser": "extended",
    "query parser fn": [Function],
    "subdomain offset": 2,
    "trust proxy": false,
    "trust proxy fn": [Function],
    "view": [Function],
    "views": "/Users/kvartborg/Code/specla/specla/test/tmp/views",
    "x-powered-by": true,
  },
}
`;

exports[`Should export properties 2`] = `"/"`;

exports[`Should proxy .all to the express instance 1`] = `
Layer {
  "handle": [Function],
  "keys": Array [],
  "name": "bound dispatch",
  "params": undefined,
  "path": undefined,
  "regexp": /\\^all\\\\/\\?\\$/i,
  "route": Route {
    "methods": Object {
      "acl": true,
      "bind": true,
      "checkout": true,
      "connect": true,
      "copy": true,
      "delete": true,
      "get": true,
      "head": true,
      "link": true,
      "lock": true,
      "m-search": true,
      "merge": true,
      "mkactivity": true,
      "mkcalendar": true,
      "mkcol": true,
      "move": true,
      "notify": true,
      "options": true,
      "patch": true,
      "post": true,
      "propfind": true,
      "proppatch": true,
      "purge": true,
      "put": true,
      "rebind": true,
      "report": true,
      "search": true,
      "subscribe": true,
      "trace": true,
      "unbind": true,
      "unlink": true,
      "unlock": true,
      "unsubscribe": true,
    },
    "path": "all",
    "stack": Array [
      Layer {
        "handle": [Function],
        "keys": Array [],
        "method": "acl",
        "name": "allCallback",
        "params": undefined,
        "path": undefined,
        "regexp": /\\^\\\\/\\?\\$/i,
      },
      Layer {
        "handle": [Function],
        "keys": Array [],
        "method": "bind",
        "name": "allCallback",
        "params": undefined,
        "path": undefined,
        "regexp": /\\^\\\\/\\?\\$/i,
      },
      Layer {
        "handle": [Function],
        "keys": Array [],
        "method": "checkout",
        "name": "allCallback",
        "params": undefined,
        "path": undefined,
        "regexp": /\\^\\\\/\\?\\$/i,
      },
      Layer {
        "handle": [Function],
        "keys": Array [],
        "method": "connect",
        "name": "allCallback",
        "params": undefined,
        "path": undefined,
        "regexp": /\\^\\\\/\\?\\$/i,
      },
      Layer {
        "handle": [Function],
        "keys": Array [],
        "method": "copy",
        "name": "allCallback",
        "params": undefined,
        "path": undefined,
        "regexp": /\\^\\\\/\\?\\$/i,
      },
      Layer {
        "handle": [Function],
        "keys": Array [],
        "method": "delete",
        "name": "allCallback",
        "params": undefined,
        "path": undefined,
        "regexp": /\\^\\\\/\\?\\$/i,
      },
      Layer {
        "handle": [Function],
        "keys": Array [],
        "method": "get",
        "name": "allCallback",
        "params": undefined,
        "path": undefined,
        "regexp": /\\^\\\\/\\?\\$/i,
      },
      Layer {
        "handle": [Function],
        "keys": Array [],
        "method": "head",
        "name": "allCallback",
        "params": undefined,
        "path": undefined,
        "regexp": /\\^\\\\/\\?\\$/i,
      },
      Layer {
        "handle": [Function],
        "keys": Array [],
        "method": "link",
        "name": "allCallback",
        "params": undefined,
        "path": undefined,
        "regexp": /\\^\\\\/\\?\\$/i,
      },
      Layer {
        "handle": [Function],
        "keys": Array [],
        "method": "lock",
        "name": "allCallback",
        "params": undefined,
        "path": undefined,
        "regexp": /\\^\\\\/\\?\\$/i,
      },
      Layer {
        "handle": [Function],
        "keys": Array [],
        "method": "m-search",
        "name": "allCallback",
        "params": undefined,
        "path": undefined,
        "regexp": /\\^\\\\/\\?\\$/i,
      },
      Layer {
        "handle": [Function],
        "keys": Array [],
        "method": "merge",
        "name": "allCallback",
        "params": undefined,
        "path": undefined,
        "regexp": /\\^\\\\/\\?\\$/i,
      },
      Layer {
        "handle": [Function],
        "keys": Array [],
        "method": "mkactivity",
        "name": "allCallback",
        "params": undefined,
        "path": undefined,
        "regexp": /\\^\\\\/\\?\\$/i,
      },
      Layer {
        "handle": [Function],
        "keys": Array [],
        "method": "mkcalendar",
        "name": "allCallback",
        "params": undefined,
        "path": undefined,
        "regexp": /\\^\\\\/\\?\\$/i,
      },
      Layer {
        "handle": [Function],
        "keys": Array [],
        "method": "mkcol",
        "name": "allCallback",
        "params": undefined,
        "path": undefined,
        "regexp": /\\^\\\\/\\?\\$/i,
      },
      Layer {
        "handle": [Function],
        "keys": Array [],
        "method": "move",
        "name": "allCallback",
        "params": undefined,
        "path": undefined,
        "regexp": /\\^\\\\/\\?\\$/i,
      },
      Layer {
        "handle": [Function],
        "keys": Array [],
        "method": "notify",
        "name": "allCallback",
        "params": undefined,
        "path": undefined,
        "regexp": /\\^\\\\/\\?\\$/i,
      },
      Layer {
        "handle": [Function],
        "keys": Array [],
        "method": "options",
        "name": "allCallback",
        "params": undefined,
        "path": undefined,
        "regexp": /\\^\\\\/\\?\\$/i,
      },
      Layer {
        "handle": [Function],
        "keys": Array [],
        "method": "patch",
        "name": "allCallback",
        "params": undefined,
        "path": undefined,
        "regexp": /\\^\\\\/\\?\\$/i,
      },
      Layer {
        "handle": [Function],
        "keys": Array [],
        "method": "post",
        "name": "allCallback",
        "params": undefined,
        "path": undefined,
        "regexp": /\\^\\\\/\\?\\$/i,
      },
      Layer {
        "handle": [Function],
        "keys": Array [],
        "method": "propfind",
        "name": "allCallback",
        "params": undefined,
        "path": undefined,
        "regexp": /\\^\\\\/\\?\\$/i,
      },
      Layer {
        "handle": [Function],
        "keys": Array [],
        "method": "proppatch",
        "name": "allCallback",
        "params": undefined,
        "path": undefined,
        "regexp": /\\^\\\\/\\?\\$/i,
      },
      Layer {
        "handle": [Function],
        "keys": Array [],
        "method": "purge",
        "name": "allCallback",
        "params": undefined,
        "path": undefined,
        "regexp": /\\^\\\\/\\?\\$/i,
      },
      Layer {
        "handle": [Function],
        "keys": Array [],
        "method": "put",
        "name": "allCallback",
        "params": undefined,
        "path": undefined,
        "regexp": /\\^\\\\/\\?\\$/i,
      },
      Layer {
        "handle": [Function],
        "keys": Array [],
        "method": "rebind",
        "name": "allCallback",
        "params": undefined,
        "path": undefined,
        "regexp": /\\^\\\\/\\?\\$/i,
      },
      Layer {
        "handle": [Function],
        "keys": Array [],
        "method": "report",
        "name": "allCallback",
        "params": undefined,
        "path": undefined,
        "regexp": /\\^\\\\/\\?\\$/i,
      },
      Layer {
        "handle": [Function],
        "keys": Array [],
        "method": "search",
        "name": "allCallback",
        "params": undefined,
        "path": undefined,
        "regexp": /\\^\\\\/\\?\\$/i,
      },
      Layer {
        "handle": [Function],
        "keys": Array [],
        "method": "subscribe",
        "name": "allCallback",
        "params": undefined,
        "path": undefined,
        "regexp": /\\^\\\\/\\?\\$/i,
      },
      Layer {
        "handle": [Function],
        "keys": Array [],
        "method": "trace",
        "name": "allCallback",
        "params": undefined,
        "path": undefined,
        "regexp": /\\^\\\\/\\?\\$/i,
      },
      Layer {
        "handle": [Function],
        "keys": Array [],
        "method": "unbind",
        "name": "allCallback",
        "params": undefined,
        "path": undefined,
        "regexp": /\\^\\\\/\\?\\$/i,
      },
      Layer {
        "handle": [Function],
        "keys": Array [],
        "method": "unlink",
        "name": "allCallback",
        "params": undefined,
        "path": undefined,
        "regexp": /\\^\\\\/\\?\\$/i,
      },
      Layer {
        "handle": [Function],
        "keys": Array [],
        "method": "unlock",
        "name": "allCallback",
        "params": undefined,
        "path": undefined,
        "regexp": /\\^\\\\/\\?\\$/i,
      },
      Layer {
        "handle": [Function],
        "keys": Array [],
        "method": "unsubscribe",
        "name": "allCallback",
        "params": undefined,
        "path": undefined,
        "regexp": /\\^\\\\/\\?\\$/i,
      },
    ],
  },
}
`;

exports[`Should proxy .delete to the express instance 1`] = `
Layer {
  "handle": [Function],
  "keys": Array [],
  "name": "bound dispatch",
  "params": undefined,
  "path": undefined,
  "regexp": /\\^delete\\\\/\\?\\$/i,
  "route": Route {
    "methods": Object {
      "delete": true,
    },
    "path": "delete",
    "stack": Array [
      Layer {
        "handle": [Function],
        "keys": Array [],
        "method": "delete",
        "name": "deleteCallback",
        "params": undefined,
        "path": undefined,
        "regexp": /\\^\\\\/\\?\\$/i,
      },
    ],
  },
}
`;

exports[`Should proxy .engine to the express instance 1`] = `
Object {
  ".pug": [Function],
}
`;

exports[`Should proxy .get (http) to the express instance 1`] = `
Layer {
  "handle": [Function],
  "keys": Array [],
  "name": "bound dispatch",
  "params": undefined,
  "path": undefined,
  "regexp": /\\^get\\\\/\\?\\$/i,
  "route": Route {
    "methods": Object {
      "get": true,
    },
    "path": "get",
    "stack": Array [
      Layer {
        "handle": [Function],
        "keys": Array [],
        "method": "get",
        "name": "getCallback",
        "params": undefined,
        "path": undefined,
        "regexp": /\\^\\\\/\\?\\$/i,
      },
    ],
  },
}
`;

exports[`Should proxy .on method to the express instance 1`] = `
Object {
  "mount": Array [
    [Function],
    [Function],
  ],
}
`;

exports[`Should proxy .param to the express instance 1`] = `
Object {
  "id": Array [
    [Function],
  ],
}
`;

exports[`Should proxy .post to the express instance 1`] = `
Layer {
  "handle": [Function],
  "keys": Array [],
  "name": "bound dispatch",
  "params": undefined,
  "path": undefined,
  "regexp": /\\^post\\\\/\\?\\$/i,
  "route": Route {
    "methods": Object {
      "post": true,
    },
    "path": "post",
    "stack": Array [
      Layer {
        "handle": [Function],
        "keys": Array [],
        "method": "post",
        "name": "deleteCallback",
        "params": undefined,
        "path": undefined,
        "regexp": /\\^\\\\/\\?\\$/i,
      },
    ],
  },
}
`;

exports[`Should proxy .put to the express instance 1`] = `
Layer {
  "handle": [Function],
  "keys": Array [],
  "name": "bound dispatch",
  "params": undefined,
  "path": undefined,
  "regexp": /\\^put\\\\/\\?\\$/i,
  "route": Route {
    "methods": Object {
      "put": true,
    },
    "path": "put",
    "stack": Array [
      Layer {
        "handle": [Function],
        "keys": Array [],
        "method": "put",
        "name": "deleteCallback",
        "params": undefined,
        "path": undefined,
        "regexp": /\\^\\\\/\\?\\$/i,
      },
    ],
  },
}
`;

exports[`Should proxy .route method to express instance 1`] = `
Layer {
  "handle": [Function],
  "keys": Array [],
  "name": "bound dispatch",
  "params": undefined,
  "path": undefined,
  "regexp": /\\^\\\\/test\\\\/\\?\\$/i,
  "route": Route {
    "methods": Object {
      "_all": true,
      "get": true,
    },
    "path": "/test",
    "stack": Array [
      Layer {
        "handle": [Function],
        "keys": Array [],
        "method": undefined,
        "name": "all",
        "params": undefined,
        "path": undefined,
        "regexp": /\\^\\\\/\\?\\$/i,
      },
      Layer {
        "handle": [Function],
        "keys": Array [],
        "method": "get",
        "name": "get",
        "params": undefined,
        "path": undefined,
        "regexp": /\\^\\\\/\\?\\$/i,
      },
    ],
  },
}
`;
